### PR TITLE
Wicker Size Property

### PR DESCRIPTION
**Pull request by isaak-willett**

### CODE REVIEW

**Comment by isaak-willett** [comment]
I'm going to add some descriptions above, this property runs like trash because of initial wicker design problems. There is no overarching location set for column files so we have to parse every row :(

File: wicker/core/datasets.py [hunk 1]
```diff
@@ -124,3 +128,74 @@ def __getitem__(self, idx: int) -> Dict[str, Any]:
...
             self.schema(),
         )
+
+    def _get_parquet_dir_size(self) -> int:
+        """Get the arrow path and find all the files within, count their bytes
+
+        Returns:
+            int: bytes in parquet directory
+        """
+        # bytes size of arrow table not bytes in arrow table
```
Comment (aalavian) [inline wicker/core/datasets.py:138]: nit: clarify this comment?

File: wicker/core/datasets.py [hunk 2]
```diff
@@ -124,3 +128,74 @@ def __getitem__(self, idx: int) -> Dict[str, Any]:
...
+        for col_name in schema.get_all_column_names():
+            if schema.get_column(col_name).is_heavy_pointer:
+                heavy_pointer_cols.append(col_name)
+
+        # Each individual row only knows which column file it goes to, so we have to
+        # neccesarily parse all rows :( to get the column files. This should be cached
+        # as metadata but that would require re-curating the datasets.
+        buckets_keys = set()
+        worker = ShuffleWorker(storage=self._storage)
+        for heavy_pntr_col in heavy_pointer_cols:
```
Comment (aalavian) [inline wicker/core/datasets.py:172]: Would we benefit from parallelizing this loop?

**Review by aalavian — APPROVED** [review]
Thanks @isaak-willett , LGTM.

File: wicker/core/datasets.py (hunk 2 above)
Comment (isaak-willett) [reply wicker/core/datasets.py:172]: I ended up adding parallelization but I haven't pushed it up yet. So yes it does benefit

File: wicker/core/datasets.py (hunk 1 above)
Comment (isaak-willett) [reply wicker/core/datasets.py:139]: Clarified

File: wicker/schema/serialization.py [hunk 3]
```diff
@@ -149,7 +149,32 @@ def _loads_base_types(
...
         raise WickerSchemaException(f"Unhandled _l5ml_metatype for avro bytes type: {l5ml_metatype}")
+    elif type_ == "record":
+        schema_fields = []
+        for schema_field in schema_dict["fields"]:
+            loaded_field = _loads_base_types(schema_field["type"], True, schema_field)
+            schema_fields.append(loaded_field)
+        return schema.RecordField(
+            fields=schema_fields,
+            name=schema_dict["name"],
+        )
```
Comment (isaak-willett) [inline wicker/schema/serialization.py:177]: Had to add in these basic field parsers, for some reason they never got added...

**Pull request merged**